src/waltham/waltham-connection.h

```diff
@@ -373,6 +354,83 @@ wth_connection_get_protocol_error(struct wth_connection *conn,
 				  const char **interface,
 				  uint32_t *object_id);
 
+/** Return a registry using the internal wth_display
+ *
+ * \param conn The Waltham connection.
+ * \return The registry obtained.
+ *
+ * The returned registry can then be used to retrieve other global
+ * objects, using its usual interface.
+ *
+ * \memberof wth_connection
+ * \client_api
+ */
+struct wthp_registry *
+wth_connection_get_registry(struct wth_connection *conn);
+
+/** Return a callback using the internal wth_display
+ *
+ * \param conn The Waltham connection.
+ * \return The callback obtained.
+ *
+ * The returned callback will emit a 'done' event once the related
+ * request is done.
+ *
+ * \memberof wth_connection
+ * \client_api
+ */
+struct wthp_callback *
+wth_connection_sync(struct wth_connection *conn);
+
+/** Set the callback to be called once a client requests a registry
+ *
+ * \param conn The Waltham connection.
+ * \param callback The registry callback.
+ * \param user_pointer Any data passed by the client.
+ *
+ * The callback will be stored inside of the connection, and will be
+ * called each time a client requests a registry.
+ *
+ * \memberof wth_connection
+ * \server_api
+ */
+void
+wth_connection_set_registry_callback(struct wth_connection *conn,
+                                     void (*callback)(struct wthp_registry *, void *),
+                                     void *user_pointer);
+
+/** Send an error on a specific object
+ *
+ * \param conn The Waltham connection.
+ * \param obj A Waltham object.
+ * \param code The error code.
+ * \param str The error message.
+ *
+ * This function wraps the wth_display_send_error display function, so
+ * the server doesn’t have to use this private API.
```

Comment:
The documentation here tells very little to the user of this library, and at the same time it is a hint that maybe this function should not be public as I noted before.